test/units/testsuite-10.service

```diff
@@ -4,4 +4,11 @@ Description=TEST-10-ISSUE-2467
 [Service]
 ExecStartPre=rm -f /failed /testok
 Type=oneshot
-ExecStart=sh -e -x -c 'rm -f /tmp/nonexistent; systemctl start test10.socket; printf x >test.file; socat -t20 OPEN:test.file UNIX-CONNECT:/run/test.ctl; >/testok'
+ExecStart=rm -f /tmp/nonexistent
+ExecStart=systemctl start test10.socket
+ExecStart=sh -x -c 'printf x >test.file'
+ExecStart=-socat -T20 OPEN:test.file UNIX-CONNECT:/run/test.ctl
+ExecStart=sleep 10
+ExecStart=sh -x -c 'test "$(systemctl show test10.socket -P ActiveState)" = failed'
+ExecStart=sh -x -c 'test "$(systemctl show test10.socket -P Result)" = trigger-limit-hit'
```

Comment:
Is it not necessary to escape `$` with `$$`? It seems all `$` in units under test-execute are escaped.

Reply:
also within single quotes ' ' ?

Reply:
Not sure if it is necessary or not, but e.g. test/test-execute/exec-ambientcapabilities.service uses `$$`.

cc @keszybz

Reply:
It seems both variants work, `'$$()'` and `'$()'`, at least they seem to produce the same result. I looked at the systemd.service man page (COMMAND LINE) but didn't find something conclusive in that regard.
I have no problems changing it though, if you think this is the more canonical way of doing things.

I'd still be interested, if `'$()'` is supposed to work of if this is just an implementation glitch.

Reply:
OK. I do not have any strong opinion about `'$$()'` or `'$()'` if they both work fine. Let's wait for other reviewers.

Reply:
if it works, ship it!

Reply:
@poettering I'm curious: Can you shed some light on the `'$$()'` vs `'$()'` situation?